src/reports_updater.rs

```diff
@@ -180,22 +350,53 @@ impl TcnDaoImpl {
 impl TcnDao for TcnDaoImpl {
     fn all(&self) -> Result<Vec<ObservedTcn>, ServicesError> {
         self.db
-            .query("select tcn, contact_time from tcn", NO_PARAMS, |row| {
-                Self::to_tcn(row)
-            })
+            .query(
+                "select tcn, contact_start, contact_end, min_distance from tcn",
+                NO_PARAMS,
+                |row| Self::to_tcn(row),
+            )
             .map_err(ServicesError::from)
     }
 
-    fn save(&self, observed_tcn: &ObservedTcn) -> Result<(), ServicesError> {
-        let tcn_str = hex::encode(observed_tcn.tcn.0);
+    fn find_tcns(
+        &self,
+        _with: Vec<TemporaryContactNumber>,
+    ) -> Result<Vec<ObservedTcn>, ServicesError> {
+        // TODO: IN query
```

Comment:
I'll do this in a separate PR. `self.all` is a valid substitute, as the TCNs are filtered programmatically, so this is just a performance optimization, and it's more important to start testing with the apps.